### DISCUSSION TITLE
Can't get dirtree to show up

### DISCUSSION THREAD

**Discussion by pinpox** [srid/neuron#601] [General]
Hey, this might be more of a question than an issue but I couldn't figure it out form the documentation:

I'm trying to get a the `dirtree` plugin to work, if I understand correctly it should tag (sub-)folders according to their name. 

Here is my simple example: https://github.com/pinpox/zettelkasten

It containes two subfolders (categories) with 2 posts each, how do I show that correctly in the `index.md`? It looks like this currently (no posts shown): 
![image](https://user-images.githubusercontent.com/1719781/115224666-71c03200-a10d-11eb-9fc1-a951fd411260.png)

How would I make the main `index.md` page show some predefined header + the cathegories with a list of all files in them (or maybe the last 10 of them)?

**Comment by srid** [comment]
> `[[z:zettels?tag=Motorcycles]]`

I believe this should instead be:

```
[[z:zettels?tag=root/Motorcycles]]
```

The dirtree plugin prefixes every hierarchical tag it generates with `root/` (so as to avoid conflict with user's tags).

**Reply by srid** [reply]
@pinpox Note: you can also try `[[z:zettels?tag=root/*&grouped]]` to display all of them grouped by the directory name. If you use `**` instead of `*`, it will recurse.

**Reply by pinpox** [reply]
@srid That doesn't seem to work either, I changed the file like this:
https://github.com/pinpox/zettelkasten/blob/022ea6f937602b8d6d60d387f4f62dbe2a8660d7/index.md

Result:
![image](https://user-images.githubusercontent.com/1719781/115524416-66e0db00-a28e-11eb-932d-4ef00a8c2527.png)


EDIT: using `[[z:zettels?tag=root/**&grouped]]` (with two `**`) doesn't make any difference either

EDIT2: Actually it seems to not even pick up the posts at all. Browsing `http://localhost:8080/Motorcycles` should show a listing, but it gives me `File not found`

**Reply by srid** [reply]
That's very strange, because I can't reproduce it on my machine, using neuron 1.9.27.2. [Using your notebook](https://github.com/pinpox/zettelkasten/commit/022ea6f937602b8d6d60d387f4f62dbe2a8660d7), I see it working (see screenshot below). What OS? How are you running neuron? Can you paste the output of neuron command? I imagine it is throwing some errors ...

![image](https://user-images.githubusercontent.com/3998/115559260-8d4c4980-a281-11eb-8b20-1d6d7f0f560a.png)

**Reply by srid** [reply]
Oh, I see that you are using version 1 of neuron (the hint was the red bar on your screenshot). `dirtree` is only available in the upcoming version 2, which can be installed using the instructions here: https://neuron.zettel.page/install

Just run `nix-env -if https://github.com/srid/neuron/archive/master.tar.gz` again, and it will upgrade your neuron.

**Reply by pinpox** [reply]
I just installed neuron-notes from nixpkgs declaratively. Sorry, didn't notice that is not the current version. Would it be possible to update the package? Even though I can still install it from Github, it would be nice to just manage it like all the other packages.

**Reply by pinpox** [reply]
Just to be sure: What version is working for you (giving you the screenshot)

**Reply by srid** [reply]
1.9.27.2

Try this approach https://neuron.zettel.page/install-declarative

**Reply by pinpox** [reply]
Thank you! I can confirm, that the newer version fixed the problem. Would be nice to have it updated in nixpkgs aswell, I can try to submit a PR but I don't know much about haskell (yet)

**Reply by srid** [reply]
I do plan cut a 2.0 release, but it requires some work ahead. https://github.com/srid/neuron/issues?q=is%3Aopen+is%3Aissue+milestone%3A2.0
